File: qc-core/src/main/java/com/eager/questioncloud/core/domain/subscribe/event/UnsubscribedEvent.java
package com.eager.questioncloud.core.domain.subscribe.event;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class UnsubscribedEvent {
    private Long creatorId;

    public static UnsubscribedEvent create(Long creatorId) {
        return new UnsubscribedEvent(creatorId);
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/subscribe/event/UnsubscribedEvent.kt
package com.eager.questioncloud.core.domain.subscribe.event

class UnsubscribedEvent(
    val creatorId: Long
) {
    companion object {
        @JvmStatic
        fun create(creatorId: Long): UnsubscribedEvent {
            return UnsubscribedEvent(creatorId)
        }
    }
}
